crates/bevy_ecs/src/component/mod.rs

```diff
@@ -39,22 +37,54 @@ impl Default for StorageType {
 }
 
 #[derive(Debug)]
-pub struct ComponentInfo {
+pub struct ComponentDescriptor {
     name: String,
-    id: ComponentId,
```

Comment:
I think the splitting of `ComponentDescriptor` out of `ComponentInfo` would be a good candidate for landing separately to make this PR easier to review. It may make sense even if this PR doesn't get merged.